core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -199,21 +212,31 @@ private void initializeCatalogTables() throws InterruptedException, SQLException
         });
   }
 
-  private void updateCatalogTables(Connection connection) throws SQLException {
-    DatabaseMetaData dbMeta = connection.getMetaData();
-    ResultSet typeColumn =
-        dbMeta.getColumns(null, null, JdbcUtil.CATALOG_TABLE_VIEW_NAME, JdbcUtil.RECORD_TYPE);
-    if (typeColumn.next()) {
-      LOG.debug("{} already supports views", JdbcUtil.CATALOG_TABLE_VIEW_NAME);
-    } else {
-      if (PropertyUtil.propertyAsBoolean(
-          catalogProperties, JdbcUtil.ADD_VIEW_SUPPORT_PROPERTY, false)) {
-        connection.prepareStatement(JdbcUtil.UPDATE_CATALOG_SQL).execute();
-      } else {
-        LOG.warn(VIEW_WARNING_LOG_MESSAGE);
-        schemaVersion = JdbcUtil.SchemaVersion.V0;
-      }
-    }
+  private void updateSchemaIfRequired() throws SQLException, InterruptedException {
+    connections.run(
+        conn -> {
+          DatabaseMetaData dbMeta = conn.getMetaData();
+          ResultSet typeColumn =
+              dbMeta.getColumns(null, null, JdbcUtil.CATALOG_TABLE_VIEW_NAME, JdbcUtil.RECORD_TYPE);
+          if (typeColumn.next()) {
+            LOG.debug("{} already supports views", JdbcUtil.CATALOG_TABLE_VIEW_NAME);
+            schemaVersion = JdbcUtil.SchemaVersion.V1;
+            return true;
+          } else {
+            if (PropertyUtil.propertyAsString(
+                    catalogProperties,
+                    JdbcUtil.SCHEMA_VERSION_PROPERTY,
+                    JdbcUtil.SchemaVersion.V0.name())
+                .equals(JdbcUtil.SchemaVersion.V1.name())) {
```

Comment:
nit: equalsIgnoreCase maybe here?

Reply:
Yeah, good idea, just in case the user does a type :)